core-to-plc/src/Language/Plutus/CoreToPLC.hs

```diff
@@ -101,6 +101,9 @@ conversionFail = (throwError . NoContext . ConversionError) <=< sdToTxt
 unsupported :: (MonadError ConvError m, MonadReader ConvertingContext m) => GHC.SDoc -> m a
 unsupported = (throwError . NoContext . UnsupportedError) <=< sdToTxt
 
+valueRestriction :: (MonadError ConvError m, MonadReader ConvertingContext m) => GHC.SDoc -> m a
+valueRestriction = (throwError . NoContext . ValueRestrictionError) <=< sdToTxt
+
 freeVariable :: (MonadError ConvError m, MonadReader ConvertingContext m) => GHC.SDoc -> m a
 freeVariable = (throwError . NoContext . FreeVariableError) <=< sdToTxt
```

Comment:
`(throwError . NoContext . err) <=< sdToTxt` seems like a pattern that can be abstracted.